_studio/mfx_lib/decode/vp8/src/mfx_vp8_dec_decode_hw.cpp

```diff
@@ -794,7 +817,7 @@ mfxStatus VideoDECODEVP8_HW::DecodeFrameCheck(mfxBitstream *p_bs, mfxFrameSurfac
         m_p_video_accelerator->EndFrame();
     }
 
-    sts = GetOutputSurface(pp_surface_out, p_surface_work, memId);
+    sts = FillOuputSurface(pp_surface_out, p_surface_work, memId);
     MFX_CHECK_STS(sts);
 
     (*pp_surface_out)->Data.Corrupted = 0;
```

Comment:
This also filling output surface, isn't it? I would suggest to either move everything to FillOuputSurface or  rename function to SetFrameType. I would probably go with 2nd variant.